roles/edpm_ovs/tasks/update.yml

```diff
@@ -0,0 +1,33 @@
+---
+- name: Update openvswitch packages without stopping the service
+  become: true
+  tags:
+    - update
+    - ovs
+  # Call dnf via the shell module as the ansible dnf module does not
+  # support passing --setopt options. This is required in order to disable any
+  # OVS rpm scriptlets which would stop and disable OVS services during
+  # a standard update. Further info:
+  # https://github.com/openstack-k8s-operators/dev-docs/blob/main/ovs-update.md#rpm-packaging
+  ansible.builtin.shell: >
+    dnf update -y openvswitch --setopt=tsflags=noscripts
+  register: update_ovs_result
+  changed_when: update_ovs_result.rc == 0
+  failed_when: update_ovs_result.rc != 0
+
+  # Reload of openvswitch service will reload OVS systemd services with an automatic
+  # restore of already set OpenFlows. Further info:
+  # https://github.com/openstack-k8s-operators/dev-docs/blob/main/ovs-update.md#proposal
+- name: Reload openvswitch service
+  become: true
+  tags:
+    - reload
+    - ovs
+  ansible.builtin.systemd_service:
+    name: openvswitch.service
+    daemon_reload: true
+    state: reloaded
+    masked: false
+  when:
+    - edpm_ovs_reload_after_update
+    - '"Nothing to do." not in update_ovs_result.stdout_lines'
```

Comment:
I know this is nitpicking and You don't need to address it now (if at all), but should we rely on the output message from dnf? What if someone will have different language set in the system?

Reply:
Good point about different languages. But I can not come up with a better way to detect that `dnf update` did not actually update anything. Return code is always 0, so I can not use that. 
I dont want to be checking OVS version explicitly as that will change over time ... hmm so it must be something not language dependent

Reply:
Here is a sample output of `update_ovs_result` when an update [happened](https://paste.opendev.org/show/bgbawBNqLRUadJyZ5bbI/) . 
Here is [no update](https://paste.opendev.org/show/bEtLB6oXxuJtTDMsjFE0/) , already up to date.

Some other solutions that are not language dependant I can think of.
 - When an update happens `stdout` will not contain word  `openvswitch`
 -  no update `stdout_lines` list is very short, so get its len()? Naach this seems flaky

Reply:
from `dnf` man page:

```
   Check-Update Command
       Command: check-update
       Aliases: check-upgrade

       dnf [options] check-update [--changelogs] [<package-file-spec>...]
          Non-interactively checks if updates of the specified packages are available.  If
          no <package-file-spec> is given, checks whether any updates at all are available
          for your system. DNF exit code will be 100 when there are updates available  and
          a  list  of  the  updates will be printed, 0 if not and 1 if an error occurs. If
          --changelogs option is specified, also changelog delta of packages about  to  be
          updated is printed.
```

Reply:
There's also `check_mode` attribute for Ansible DNF module that returns the availability of an update: https://docs.ansible.com/ansible/latest/collections/ansible/builtin/dnf_module.html#attribute-check_mode

Reply:
Thanks, classic RTFM on my end :). Ended up going with the `check_mode` as I prefer to use ansible modules instead of raw shell command.